faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -374,7 +395,7 @@ private FullHttpRequest newRequest(HttpMethod method, String path, JsonNode body
     return request;
   }
 
-  private CompletableFuture<FullHttpResponse> performRequest(final FullHttpRequest request) {
+  private CompletableFuture<FullHttpResponse> performRequest(final FullHttpRequest request, final Optional<Duration> requestQueryTimeout) {
```

Comment:
Was the timeout supposed also the being handle in the [client](https://github.com/fauna/faunadb-jvm/blob/ea5935b79851ce3457dec9841d3ce202a5ccac5d/faunadb-common/src/main/java/com/faunadb/common/Connection.java#L427 ) ? Currently, we are only setting the headers, right ?

Reply:
For the scope of this ticket, we're only setting the headers. If we handle it on the client side as well, we should consider the time for request/response roundtrip as well. Maybe it should be a different setting? (Arguably, clients could just timeout the future in case they want to)

The query timeout is supposed to count only the time it takes to run the query on the _server_ side. @nathaniel-may, could you confirm this is the case? :) Thanks!